### ISSUE TITLE
Possible issue with zh-tw(or any other languages that are not included in the base game)

### ISSUE THREAD

**Issue by sway2020** [algernon-A/Ploppable-RICO-Revisited#49]
From reading the translation framework code I noticed that if a language is not included in the base game(not a locale option), it will never be used(such as zh-tw). TMPE has more languages than the game does so TMPE uses an additional drop-down list in mod setting to let users manually choose what language to be used. I'll try to do something similar in Find It Fix.

**Comment by sway2020** [comment]
Btw there's a Japanese modder, mzch, who makes custom Japanese locale files for the game and I heard it is quite popular among Japanese players(Japanese isn't included in the base game). Here's the download link to the newest version of his locale work. It might be helpful for testing. https://drive.google.com/file/d/1jC-F1rrpUs3q_x6Egy4rMkPoGrcjxoLx/view?usp=sharing

He also made a steam workshop version. https://steamcommunity.com/sharedfiles/filedetails/?id=427164957

**Comment by algernon-A** [comment]
There are a couple of mods on the Workshop that add zh-tw to the base game, which this framework picks up and works with.  You are correct that a dropdown would be good, and I'm already doing some work to extend the framewok to include an easy hook for that.

**Comment by algernon-A** [comment]
Framework will be updated to include a method that will return all available translations in an ordered `string[]` array suitable for direct use for a `UIDropDown` items list, and another method that will set the language based on the selected index.  E.g:

```
myUIDropDown.items = Translations.GetList();
myUiDropDown.eventIndexChange += eventSelectedIndexChanged += (control, index) =>
{
        Translations.SelectTranslation(index);
};
```

**Comment by algernon-A** [comment]
Alright, I've just committed an updated translations framework here.  Feel free to use it for Find It - it'll make life so much easier for you.

Adding a drop-down language choice menu in your options panel is now as simple as:

`UIDropDown translationDropDown = (UIDropDown)yourPanel.AddDropdown(Translations.Translate("TRN_CHOICE"), Translations.LanguageList, Translations.Index, (value) => { Translations.Index = value; });`

(Don't forget to save, of course).

Adding the serialization/deserialization of this setting to your XML settings file is as simple as:
```
        [XmlElement("Language")]
        public string language
        {
            get
            {
                return Translations.Language;
            }
            set
            {
                Translations.Language = value;
            }
        }
```

Make sure you have these two keys in your translation files:
```
    <Translation ID="TRN_CHOICE" String="Set language" />
    <Translation ID="TRN_SYS" String="Use system language" />
```

**Comment by sway2020** [comment]
@algernon-A thanks a lot! I got a question with the implementation of settings. It seems like the SamsamTS used CO's framework to save the settings instead of having an XML file. Do you know the pros and cons of this?  Here's the code https://github.com/sway2020/CS-FindIt/blob/master/FindIt/ModInfo.cs

**Comment by algernon-A** [comment]
Fixed by 1f01b69d7dabc626a814bfa62223ca6506ae6801
